internal/credential/vault/repository_ssh_certificate_credential_library.go

```diff
@@ -150,6 +164,20 @@ func (r *Repository) UpdateSSHCertificateCredentialLibrary(ctx context.Context,
 			return nil, db.NoRowsAffected, errors.New(ctx, errors.InvalidFieldMask, op, f)
 		}
 	}
+
+	if keyTypeChange && l.KeyType == "" {
+		l.KeyType = KeyTypeEd25519
+	}
+
+	if keyTypeChange && keyBitChangeDefault {
```

Comment:
`keyTypeChange` required in this conditional? keybits could change to default but keytype doesn't resulting in invalid keybits

Reply:
Yep, good catch. I will add some logic if only the key bits change. In which case we need to look at the keytype from `origLib` below

Reply:
New tests for this case ecc3e7dcb and implementation to make those test pass 1ae357271